composition/src/normalization/normalization-factory.ts

```diff
@@ -265,6 +274,112 @@ export function normalizeSubgraph(
   return normalizationFactory.normalize(document);
 }
 
+export function validateEventSubscribetionSubjects(fieldDefinition: FieldDefinitionNode, argumentNodeValue: ConstValueNode, errorMessages: string[]): boolean {
+  if (argumentNodeValue.kind !== Kind.LIST) {
+    errorMessages.push(invalidEventSubjectsErrorMessage);
+
+    return false;
+  }
+
+  const argsNames: Array<string> = [];
+
+  for (const value of argumentNodeValue.values) {
+    if (value.kind !== Kind.STRING || value.value.length < 1) {
+      errorMessages.push(invalidEventSubjectsItemErrorMessage);
+
+      return false;
+    }
+
+    const val = value.value;
+    if (val.includes('{{')) {
+      if (!validateSubjectTemplate(fieldDefinition, val, argsNames, errorMessages)) {
+        return false;
+      }
+    }
+  }
+
+  if (fieldDefinition.arguments?.length !== undefined && argsNames.length > 0) {
+    for (const argument of fieldDefinition.arguments) {
+      if (argsNames.indexOf(argument.name.value) === -1) {
+        errorMessages.push(invalidFieldDefinitionNoArgumentTemplateErrorMessage);
+
+        return false;
+      };
+    }
+  }
+
+  if (argsNames.length === 0 && fieldDefinition.arguments?.length !== undefined) {
+    errorMessages.push(invalidFieldDefinitionNoArgumentTemplateErrorMessage);
+
+    return false;
+  }
+
+  return true
+}
+
+function validateSubjectTemplate(fieldDefinition: FieldDefinitionNode, value: string, argsSet: Array<string>, errorMessages: string[]): boolean {
```

Comment:
In the natsserver package in go, there is a method to validate a nats subject. It's likely that the nats js client has some thing similar. But if not, you could look what the go client does, and lift that.

I think the best way to validate is to match each argument template and validate it against the arguments.
After that is all clear, change each argument template to a dummy value and check whether it's a valid nats subject.